source/customization/menus.rst

```diff
@@ -1,5 +1,367 @@
+     File/Menu Name          Description
+========================   ===============
+**Main**                   Main menu
+
+**Context**                Context menu in the editing area
+
+**Find in Files**          Appears when clicking the "…" button
+                           in the :doc:`Find in Files
+                           </search_and_replace/search_and_replace_files>` panel
+
+**Side Bar**               Context menu for each node in the sidebar
+
+**Side Bar Mount Point**   Additional context menu items
+                           for the top-level nodes in the sidebar
+
+**Tab Context**            Context menu of the tab bar
+
+**Widget Context**         Context menu of input fields
+                           in all kinds of widgets,
+                           including Command Palette, Goto Anything,
+                           the find panels
+                           and those opened by APIs
+========================   ===============
+
+Additionally,
+the following four menus open
+when their column in the status bar
+is clicked:
+
+- **Encoding**
+- **Line Endings**
+- **Indentation**
+- **Syntax**
+
+.. figure:: images/statusbar_menu.gif
+   :class: shadowed
+
+   Demonstration of a status bar menu.
+
+
+.. _menu-items:
+
+Menu Items
+==========
+
+A menu item can either
+invoke a command (with arguments)
+when it is selected
+or have a sub-menu.
+
+The available properties are:
+
+- a command name,
+- arguments for the command,
+- an ID,
+- a caption,
+- a mnemonic and
+- a sub-menu.
+
+In order to function properly,
+it must provide *at least* either out of
+
+- a command name,
+- a caption and a sub-menu or
+- just a caption or
+- an ID (see below).
+
+When parsing a menu item,
+the following rules apply:
+
+#. The presence of a sub-menu
+   overrides the command and its arguments,
+   making it effectively non-existent.
+   It also overrides the :ref:`separator caption <menu-separators>`.
+
+#. If no caption is provided,
+   a caption is inferred
+   from the command's ``description`` method.
+   If neither caption nor command are provided,
+   the caption is an empty string.
+
+#. The character used for the mnemonic
+   must be contained in the item's caption.
+   Mnemonics are case-sensitive.
+
+#. Menu items referencing
+   commands that can not be found
+   are disabled.
+
+#. Menu items can be hidden or disabled
+   by the command.
+
+.. TODO add refs to Command methods
+
+
+.. _menu-separators:
+
+Separators
+**********
+
+Separators are menu items
+with the caption ``-``
+and no sub-menu.
+They are commonly used
+to group menu items with similar purposes
+or with a shared thematic.
+Separators can not call commands,
+but the presence of a sub-menu
+converts the separator item into a regular item
+with a single hyphen as its caption.
+
+For rendering,
+multiple consecutive separators are reduced to one
+and separators at the beginning or the end of a menu
+are not displayed.

```

Comment:
Could be misleading imo. If there is a separator at the very beginning of a menu with an id it is relevant and definitely not ignored. It's just not rendered.